torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -24,6 +25,38 @@ static constexpr char* kEngineCPUQueueSize =
     "local_autograd_engine_cpu_queue_size";
 static constexpr char* kNumAutogradContexts = "num_autograd_contexts";
 
+struct AccumulateGradPreHook : GraphTask::GraphTaskFunctionPreHook {
```

Comment:
Separate implementation in .h and .cpp file? Also add some docs to this class stating its purpose.

Reply:
That's not necessary. This is an internal class dedicated to `DistEngine`.

Reply:
Added comments.